packages/arb-token-bridge-ui/src/components/common/NetworkSelectionContainer.tsx

```diff
@@ -1,38 +1,45 @@
-import { useWallet } from '@arbitrum/use-wallet'
-import { Web3Provider } from '@ethersproject/providers'
 import { Popover, Transition } from '@headlessui/react'
 import Image from 'next/image'
+import { useCallback } from 'react'
+import { useNetwork, useSwitchNetwork } from 'wagmi'
+
 import {
   ChainId,
   getNetworkLogo,
   getNetworkName,
-  switchChain
+  handleSwitchNetworkError,
+  isNetwork
 } from '../../util/networks'
 
 export const NetworkSelectionContainer = ({
-  supportedNetworks,
   children
 }: {
-  supportedNetworks: ChainId[]
   children: React.ReactNode
 }) => {
-  const { provider } = useWallet()
+  const { chain } = useNetwork()
+  const { switchNetwork } = useSwitchNetwork({
+    throwForSwitchChainNotSupported: true,
+    onError: handleSwitchNetworkError
+  })
+  const supportedNetworks = isNetwork(chain?.id ?? 0).isTestnet
```

Comment:
Nice work refactoring it here. 
We can also try moving `supportedNetworks` to `network.ts` and use the same logic for supportedNetworks in this file + in `MainNetworkNotSupported.tsx` because that uses duplicate code as well.

Reply:
good catch!